### DISCUSSION TITLE
JavaScript Transformer Not Working on Inbound XML

### DISCUSSION THREAD

**Discussion by ruizhannia** [nextgenhealthcare/connect#4492] [Q&A] [answered]
I am working on a JavaScript transformer to build an HL7 ORU from an XML. I am currently working on the MSH segment with the following script:
//MSH.3 - sendingApplication
tmp['MSH']['MSH.3']['MSH.3.1'] = msg['sendingApplication'].toString();;
//MSH.4 - sendingFacility
tmp['MSH']['MSH.4']['MSH.4.1'] = msg['sendingFacility'].toString();
//MSH.4 - Practice ID
tmp['MSH']['MSH.4']['MSH.4.2'] = msg['sendingApplication'].toString();
//tmp['MSH']['MSH.4']['MSH.4.2'] = '0001';
//MSH.5 - receivingApplication
tmp['MSH']['MSH.5']['MSH.5.1'] = msg['receivingApplication'].toString();
//MSH.6 - receivingFacility
tmp['MSH']['MSH.6']['MSH.6.1'] = msg['receivingFacility'].toString();
//MSH.7 - Date/Time of message
tmp['MSH']['MSH.7']['MSH.7.1'] = DateUtil.getCurrentDate("yyyyMMddHHmmss.SSSZZ");
//MSH9.2 - Trigger Event
tmp['MSH']['MSH.9']['MSH.9.2'] = msg['documentMetadata']['triggerEventId'].toString();
//MSH.10 - messageControlID
tmp['MSH']['MSH.10']['MSH.10.1'] = msg['messageControlId'].toString();
//MSH.11 - Processing ID
tmp['MSH']['MSH.11']['MSH.11.1'] = msg['processingId'].toString();

The only transformation that is passing through is the MSH.7 for the date. I am not sure why the inbound message is not being recognized. Also, if I hardcode a segment it does get added to the HL7. I have another channel with a similar set-up and have compared the data types settings, and all other filter and transformer configurations but I am not seeing anything different from what I have set-up.

**Comment by tonygermano** [comment]
Can you share a sample inbound message? If `msg['sendingApplication']` doesn't exist, then `msg['sendingApplication'].toString()` will return an empty string.

**Reply by tonygermano** [reply]
You're going to have to format that as a code block. It doesn't look like xml. Better yet, edit your original post and put it in your question. See the link below the comment that says "Styling with Markdown is supported" for formatting options.

**Comment by ruizhannia** [comment]
This is a portion of the test XML I am working with: 
`<publishDocumentRequest>
   <version>2</version>
   <documentMetadataId>EA1287037949394O4</documentMetadataId>
   <sendingApplication>98765</sendingApplication>
   <sendingFacility>ClinicTwo</sendingFacility>
   <receivingApplication>123456</receivingApplication>
   <receivingFacility>OneClinic</receivingFacility>
   <messageControlId>11</messageControlId>
   <processingId>P</processingId>
    <DId>3212</cDId>
   <patientDemographics>
      <lastName>Test</lastName>
      <firstName>Patient</firstName>
      <dob>1970-01-01</dob>
      <gender>M</gender>
      <address>
         <city>Springfield</city>
         <state>MO</state>
         <zipCode>12345</zipCode>
      </address>
   </patientDemographics>
<publishDocumentRequest>`

**Reply by tonygermano** [reply]
I had to massage a couple of your tags to get that xml to actually load, but once I did, your code in the original post is working. I'm assuming the errors are not in the original data or the transformer would error and not run at all.

**Reply by ruizhannia** [reply]
Thank you for taking a look. I was able to identify the issue after running a check through all the configurations on the channel. It turns out there was an HL7 inbound template within the Source transformer that was causing the issue. Once I removed the template and processed a test message through the transformer on the destination worked properly. I had missed this while I was running initial testing as I did not pay attention to the fact that it had been transforming to HL7 on the source.

**Reply by tonygermano** [reply]
Please write your resolution as an Answer instead of a comment and mark it as the solution.

**Comment by ruizhannia** [comment, marked as answer]
Issue was found to be an HL7 outbound template in the Source connector. Once the template was removed, the transformer worked as expected. Since template was in the source connector once the message came to the destination transformer it was in HL7 format and therefore the values  msg variables were not recognized for that HL7.
